.github/workflows/regression-report.yml

```diff
@@ -641,10 +641,18 @@ jobs:
           )
 
           for key in "${CACHE_KEYS[@]}"; do
-            echo "Removing cache with key: $key"
-            curl --fail-with-body -X DELETE \
+            echo "Attempting to remove cache with key: $key"
+            response=$(curl -s -o /dev/null -w "%{http_code}" -X DELETE \
               -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" \
               -H "Accept: application/vnd.github.v3+json" \
-              "https://api.github.com/repos/${{ github.repository }}/actions/caches?key=$key"
+              "https://api.github.com/repos/${{ github.repository }}/actions/caches?key=$key")
+
+            if [ "$response" = "204" ]; then
+              echo "Successfully removed cache: $key"
+            elif [ "$response" = "404" ]; then
+              echo "Cache not found (already removed or never existed): $key"
+            else
+              echo "Unexpected response ($response) when trying to remove cache: $key"
```

Comment:
This is prone to do nothing if we failing to remove, due to mistake in name of cache.

Please add values to array by if condition  , example of how add new value to array https://ioflood.com/blog/bash-append-to-array/#:~:text=In%20Bash%2C%20you%20can%20append,')%20arr%2B%3D('!